daemon/device-plugin/deviceplugin.go

```diff
@@ -75,26 +75,40 @@ func (nf *nfResources) Changed() bool {
 	return changed
 }
 
+func (nf *nfResources) ListAndWatch(empty *pluginapi.Empty, stream pluginapi.DevicePlugin_ListAndWatchServer) error {
+	changed := true
+	for {
+		if changed {
+			err := nf.sendDevices(stream)
+			if err != nil {
+				return err
+			}
+		}
+		time.Sleep(5 * time.Second)
+		changed = nf.changed()
+	}
```

Comment:
Since this PR is already cleaning up the code, let's go one step further. The code in the change uses `nf.changed()` but also somehow manages changed state with a boolean (to for a change every time).  In addition, it has a bunch of side-effects saving the state to nf, but really that state is only needed to determine what to send to kube api.

I propose we rewrite all the other functions so that we can write something like this:
```
    var oldDevices []Device
    for {
        newDevices := getDevices()
        if !devicesEqual(oldDevices, newDevices) {
            if err := sendDevices(stream, newDevices); err != nil {
                return err
            }
            oldDevices = newDevices
        }
        time.Sleep(5 * time.Second)
    }
```

Reply:
Done